stripe-crypto-onramp/src/main/java/com/stripe/android/crypto/onramp/CheckoutState.kt

```diff
@@ -0,0 +1,36 @@
+package com.stripe.android.crypto.onramp
+
+import com.stripe.android.crypto.onramp.model.OnrampCheckoutResult
+import com.stripe.android.model.PaymentIntent
+
+/**
+ * Represents the checkout state with session info and current status.
+ */
+internal data class CheckoutState(
+    val onrampSessionId: String? = null,
+    val checkoutHandler: (suspend () -> String)? = null,
+    val status: Status = Status.Idle
+) {
+    internal sealed interface Status {
+        /**
+         * No checkout is currently in progress.
+         */
+        data object Idle : Status
+
+        /**
+         * Checkout is currently processing.
+         */
+        data object Processing : Status
```

Comment:
Classes are preferred over objects for compat reasons. We'll need to make similar changes elsewhere as well.

```suggestion
        class Processing internal constructor() : Status
```